lib/vector-core/src/event/metric.rs

```diff
@@ -571,35 +577,38 @@ impl MetricValue {
                     count: count2,
                     sum: sum2,
                 },
-            ) => {
-                if quantiles.len() == quantiles2.len()
-                    && quantiles
-                        .iter()
-                        .zip(quantiles2.iter())
-                        .all(|(b1, b2)| b1.upper_limit == b2.upper_limit)
-                {
-                    for (b1, b2) in quantiles.iter_mut().zip(quantiles2) {
-                        b1.value += b2.value;
-                    }
-                    *count += count2;
-                    *sum += sum2;
+            ) if quantiles.len() == quantiles2.len()
+                && quantiles
+                    .iter()
+                    .zip(quantiles2.iter())
+                    .all(|(b1, b2)| b1.upper_limit == b2.upper_limit) =>
+            {
+                for (b1, b2) in quantiles.iter_mut().zip(quantiles2) {
+                    b1.value += b2.value;
```

Comment:
This looks like we're taking two quantiles that are, say, individually representing something like `50ms at the 50th percentile`, and then doing `50 + 50`.... but I can't think of a situation where this behavior would ever be correct.

I know you're just updating the structure here, but this seems like something we at least need to document and come back around to fixing / removing / something.

Reply:
You are correct. This is a questionable practice and we should mark it for further reworking.

FWIW the structure of that code was inherited from a much earlier version, back in 2019 in fact: c112c4ac7f45e69fea312e7691566a3f9e8e3066